Makefile

```diff
@@ -340,6 +347,7 @@ LIBC_TOP_HALF_MUSL_SOURCES += \
         thread/pthread_rwlockattr_init.c \
         thread/pthread_rwlockattr_setpshared.c \
         thread/pthread_setcancelstate.c \
+        thread/pthread_setcanceltype.c \
```

Comment:
Perhaps making these additions to the multi-threaded build should be done as its own separate PR?

Reply:
Started breaking this up into smaller commits